File: Server/src/main/java/org/example/dto/category/CategoryUpdateDTO.java
package org.example.dto.category;

import jakarta.annotation.Nonnull;
import jakarta.annotation.Nullable;
import lombok.Data;
import org.springframework.web.multipart.MultipartFile;

import jakarta.validation.constraints.NotBlank;

@Data
public class CategoryUpdateDTO {
    @NotBlank(message = "Назва є обов'язковою")
    private String name;

    @NotBlank(message = "Опис є обов'язковим")
    private String description;

    @Nullable
    private MultipartFile image;
}

File: Server/src/main/java/org/example/dto/category/CategoryUpdateDTO.kt
package org.example.dto.category

import jakarta.validation.constraints.NotBlank
import org.springframework.web.multipart.MultipartFile

open class CategoryUpdateDTO {
    lateinit var name: @NotBlank(message = "Назва є обов'язковою") String
    lateinit var description: @NotBlank(message = "Опис є обов'язковим") String
    var image: MultipartFile? = null
}
